packages/platforms/webos-player/src/classes/webos-machine.ts

```diff
@@ -0,0 +1,184 @@
+  }
+
+  async removeCredentials(): Promise<void> {
+    console.log('removeCredentials');
+    return storage.removeFile({
+      file: CREDENTIALS_FILE_PATH,
+    });
+  }
+
+  async getLocation(): Promise<
+    undefined | { latitude: number; longitude: number }
+  > {
+    console.log('getLocation');
+    try {
+      const location = await new Promise<GeolocationPosition>(
+        (resolve, reject) => {
+          navigator.geolocation.getCurrentPosition(resolve, reject);
+        }
+      );
+
+      return {
+        latitude: location.coords.latitude,
+        longitude: location.coords.longitude,
+      };
+    } catch (e) {
+      return undefined;
+    }
+  }
+
+  async getTimezone(): Promise<string> {
+    console.log('getTimezone');
+    return Intl.DateTimeFormat().resolvedOptions().timeZone;
+  }
+
+  async getDeviceInfo(): Promise<DeviceInfo> {
+    console.log('getDeviceInfo');
+
+    const platformInfo = await deviceInfo.getPlatformInfo();
+    // get chromium version from user agent
+    const chromiumVersion =
+      navigator.userAgent.match(/Chrome\/([0-9.]+)/)?.[1] ?? undefined;
+    return {
+      appType: 'LG WebOS',
+      appVersion: version,
+      os: 'LG WebOS',
+      hardware: platformInfo.modelName,
+      environmentVersion: platformInfo.firmwareVersion,
+      chromiumVersion,
+      userAgent: navigator.userAgent,
+    };
+  }
+
+  /**
+   * Restart the device application.
+   */
+  async restart(): Promise<void> {
+    console.log('restart');
+    return configuration.restartApplication();
+  }
+
+  /**
+   * Quit the device application.
+   */
+  async quit(): Promise<void> {
+    console.error('Not supported on webOS.');
+  }
+
+  /**
+   * Reboot the device. This should perform a clean hardware reboot of the device.
+   *
+   */
+  async reboot(): Promise<void> {
+    console.log('reboot');
+    return power.executePowerCommand({
+      powerCommand: 'reboot',
+    });
+  }
+
+  /**
+   * Shutdown the device. This should perform a clean hardware shutdown of the device.
+   * i.e. after this method is called the device should be completely powered off.
+   */
+  async shutdown(): Promise<void> {
+    console.log('shutdown');
+    return power.executePowerCommand({
+      powerCommand: 'powerOff',
+    });
+  }
+
+  /**
+   * Updates the device's application.
+   */
+  async update(): Promise<void> {
+    console.log('update');
+
+    // First set the server properties to the correct values
+    await configuration.setServerProperty(UPDATE_CONFIG);
+
+    // Then download the application
+    await storage.upgradeApplication({
+      type: 'ipk',
+      to: 'local',
+      recovery: true,
+    });
+
+    // Reboot the device to apply the update
+    return this.reboot();
+  }
+
+  /**
+   * Updates the device's firmware.
+   */
+  async updateFirmware(): Promise<void> {
+    console.log('updateFirmware');
+
+    const url = await this.getFirmwareDownloadUrl();
+
+    await storage.downloadFirmware({
```

Comment:
Will this file be removed automatically after the upgrade?

Reply:
Yes. It is deleated automatically upon reboot. Added a comment about it